### PR TITLE
Update RPC methods: Add ContainsBTCBlock(), update GetBTCHeaderChainTip()

### PR DESCRIPTION
**Pull request by udit-gulati**
## Description

1. Adds ContainsBTCBlock() method to rpc to check if blockHash is present in lightclient or not
2. Updates GetBTCHeaderChainTip() method to add lastest block height as return value

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
    
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

### CODE REVIEW

File: btclightclient/btc_light_client.go [hunk 1]
```diff
@@ -225,11 +225,20 @@ func (lc *BTCLightClient) Status() {
 	fmt.Println(lc.btcStore.MostDifficultFork())
 }
 
+func (lc *BTCLightClient) LatestBlockHeight() int64 {
```
Comment (vuvoth) [inline btclightclient/btc_light_client.go:228]: This is not LatestBlockHeight. I think for right meaning this must be LatestFinalizedBlockHeight

File: btclightclient/btc_light_client_store.go [hunk 2]
```diff
@@ -48,6 +49,11 @@ func NewMemStore() *MemStore {
 	}
 }
 
+func (s *MemStore) IsBlockPresent(h chainhash.Hash) bool {
+	_, ok := s.lightBlockByHashMap[h]
```
Comment (vuvoth) [inline btclightclient/btc_light_client_store.go:53]: IsBlockPresent can use `LightBlockByHash` API to check block hash `h` exists in storage. 
```golang
s.LightBlockByHash(h) != nil
```

File: btclightclient/btc_light_client.go (hunk 1 above)
Comment (udit-gulati) [reply btclightclient/btc_light_client.go:228]: Then shall I also change the name in `Store` interface?

Comment (vuvoth) [reply btclightclient/btc_light_client.go:228]: Yes. Please!

File: btclightclient/btc_light_client_store.go [hunk 3]
```diff
@@ -48,6 +49,11 @@ func NewMemStore() *MemStore {
 	}
 }
 
+func (s *MemStore) IsBlockPresent(h chainhash.Hash) bool {
```
Comment (vuvoth) [inline btclightclient/btc_light_client_store.go:52]: I think we can use this logic in the light client file. We don't need to add one more interface for this. 
However, if you see this is cleaner, please let me know. I am good at all points.

File: btclightclient/btc_light_client.go [hunk 4]
```diff
@@ -225,9 +225,19 @@ func (lc *BTCLightClient) Status() {
 	fmt.Println(lc.btcStore.MostDifficultFork())
 }
 
-func (lc *BTCLightClient) LatestBlockHash() chainhash.Hash {
-	latestBlockHash := lc.btcStore.LatestCheckPoint().Header.BlockHash()
-	return latestBlockHash
+func (lc *BTCLightClient) FinalizedBlockHeight() int64 {
```
Comment (vuvoth) [inline btclightclient/btc_light_client.go:228]:
```suggestion
func (lc *BTCLightClient) LatestFinalizedBlockHeight() int64 {
```

File: btclightclient/btc_light_client.go [hunk 5]
```diff
@@ -225,9 +225,19 @@ func (lc *BTCLightClient) Status() {
...
+	latestFinalizedBlockHeight := lc.btcStore.FinalizedHeight()
+	return latestFinalizedBlockHeight
+}
+
+func (lc *BTCLightClient) FinalizedBlockHash() chainhash.Hash {
```
Comment (vuvoth) [inline btclightclient/btc_light_client.go:233]:
```suggestion
func (lc *BTCLightClient) LatestFinalizedBlockHash() chainhash.Hash {
```

File: btclightclient/btc_light_client_store.go [hunk 6]
```diff
@@ -11,7 +11,7 @@ const MaxForkAge = 8
 
 type Store interface {
 	LightBlockAtHeight(int64) *LightBlock
-	LatestHeight() int64
+	FinalizedHeight() int64
```
Comment (vuvoth) [inline btclightclient/btc_light_client_store.go:14]:
```suggestion
	LatestFinalizedHeight() int64
```

**Review by vuvoth — APPROVED** [review]
LGTM! 
Please check my comments!

**Pull request merged**